x-pack/plugins/security_solution/public/common/containers/use_global_time/index.tsx

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useCallback, useState, useEffect, useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { inputsSelectors } from '../../store';
+import { inputsActions } from '../../store/actions';
+
+export const useGlobalTime = () => {
+  const dispatch = useDispatch();
+  const { from, to } = useSelector(inputsSelectors.globalTimeRangeSelector);
+  const [isInitializing, setIsInitializing] = useState(true);
+
+  const setQuery = useCallback(
+    ({
+      id,
+      inspect,
+      loading,
+      refetch,
+    }: Pick<
```

Comment:
Why we use `Pick` here? Expect I am missing something, I think it is better to define an interface.